File: src/models/ff/ff_rangeruler.rs
use super::{ff_passrange::FfPassRangeEnum};
use serde::{Deserialize, Serialize};

#[derive(Default, Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
pub struct FfRangeRuler {
    quick_pass: i32,
    short_pass: i32,
    long_pass: i32,
    long_bomb: i32
}

impl FfRangeRuler {
	pub fn new() -> FfRangeRuler {
        FfRangeRuler {
            quick_pass: 3,
            short_pass: 7,
            long_pass: 10,
            long_bomb: 13
        }
    }    

    pub fn get_pass_range(self, distance: i32) -> FfPassRangeEnum {
		
        let rr = FfRangeRuler::new();

        if distance > rr.long_bomb {
			return FfPassRangeEnum::OutOfRange;
		} else if distance > rr.long_pass {
			return FfPassRangeEnum::LongBomb;
		} else if distance > rr.short_pass {
			return FfPassRangeEnum::LongPass;
		} else if distance > rr.quick_pass {
			return FfPassRangeEnum::ShortPass;
		} else {
			return FfPassRangeEnum::QuickPass;
		}		
	}
}

#[cfg(test)]
mod tests {
    use crate::models::ff::{ff_passrange::FfPassRangeEnum, ff_rangeruler::FfRangeRuler};


    #[test]
    fn reset_get_pass_range() {
        let r_r = FfRangeRuler::new();

        assert_eq!(r_r.get_pass_range(2), FfPassRangeEnum::QuickPass);
    }
}

File: java/src/main/java/com/foul/gameserver/games/FfGame/FfRangeRuler.java
package com.foul.gameserver.games.FfGame;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@Getter
@EqualsAndHashCode
@ToString
public final class FfRangeRuler {

    Logger logger = LoggerFactory.getLogger(FfRangeRuler.class);

    int quickPass;
    int shortPass;
    int longPass;
    int longBomb;
   
    FfRangeRuler() {
        this.quickPass = 3;
        this.shortPass = 7;
        this.longPass = 10;
        this.longBomb = 13;
    }

    FfPassRangeEnum getPassRange(int distance){
		
        FfRangeRuler rr = new FfRangeRuler();

        if( distance > rr.longBomb ) {
			return FfPassRangeEnum.OutOfRange;
		} else if( distance > rr.longPass ) {
			return FfPassRangeEnum.LongBomb;
		} else if( distance > rr.shortPass ) {
			return FfPassRangeEnum.LongPass;
		} else if( distance > rr.quickPass ) {
			return FfPassRangeEnum.ShortPass;
		} else {
			return FfPassRangeEnum.QuickPass;
		}		
	}
}
